src/liesel/model/nodes.py

```diff
@@ -938,6 +939,195 @@ def all_input_nodes(self) -> tuple[Node, ...]:
+
+        # Helper function to get and validate bijector for a parameter
+        def get_bijector(param_name: str) -> jb.Bijector | None:
+            """
+            Gets the constraining bijector for a parameter.
+
+            Returns None if the parameter doesn't need transformation.
+            Raises RuntimeError if invalid or bijector retrieval fails.
+            """
+            prop = param_props.get(param_name)
+
+            if not hasattr(prop, 'default_constraining_bijector_fn'):
+                raise RuntimeError(
+                    f"Parameter property for '{param_name}' of "
+                    f"{self.distribution.__name__} does not have "
+                    f"'default_constraining_bijector_fn' attribute. This may "
+                    f"indicate an issue with the TFP distribution or version. "
+                    f"Either use a distribution that supports "
+                    f"parameter_properties() or manually transform parameters "
+                    f"with .transform()."
+                )
+
+            try:
+                bijector = prop.default_constraining_bijector_fn()  # type: ignore
+            except Exception as e:
+                raise RuntimeError(
+                    f"Error getting bijector for parameter '{param_name}' of "
+                    f"{self.distribution.__name__}: {e}"
+                ) from e
+
+            if bijector is None:
+                raise RuntimeError(
+                    f"Expected a bijector or BIJECTOR_NOT_IMPLEMENTED for "
+                    f"parameter '{param_name}' of {self.distribution.__name__}, "
+                    f"but got None. If no default bijector should be provided for "
+                    f"this parameter, the return should be the "
+                    f"BIJECTOR_NOT_IMPLEMENTED method instead."
+                )
+
+            bijector_type_name = type(bijector).__name__
+            if bijector_type_name == 'BIJECTOR_NOT_IMPLEMENTED':
+                # TFP's way of indicating no default bijector exists
+                return None
+
+            return bijector
+
+        # Helper function to process a single parameter
+        def process_param(param_name: str, input_node: Node) -> None:
+            """
+            Processes a parameter input node and adds it to transformable if applicable.
+            """
+            # Only VarValue nodes (from Var inputs) can be auto-transformed
+            if not isinstance(input_node, VarValue):
+                return
+
+            param_var = input_node.var
+
+            if param_var is None or not param_var.parameter:
+                # Not a parameter variable, skip
+                return
+
+            if param_var.weak:
+                # Weak variables are already computed/transformed, skip silently
+                return
+
+            bijector = get_bijector(param_name)
+
+            if bijector is not None:
+                transformable[param_name] = (param_var, bijector)
```

Comment:
If I see it correctly, this helper function mutates the transformable dictionary directly. This is fine and works, but it would be easier to maintain if it would act a bit more like a "pure" function, so that the changing of the transformable dictionary happens on the top level of this method.